src/openrct2/paint/VirtualFloor.cpp

```diff
@@ -347,30 +362,30 @@ void virtual_floor_paint(paint_session* session)
     // Edges which are internal to objects (i.e., the tile on both sides
     //  is occupied/lit) are not rendered to provide visual clarity.
     uint8_t dullEdges = 0xF & ~occupiedEdges & ~litEdges;
-    uint8_t paintEdges = (weAreOccupied || weAreLit) ? ~dullEdges : 0xF;
+    uint8_t paintEdges = ((weAreOccupied || weAreLit) && weAreOwned) ? ~dullEdges : 0xF;
 
-    if (paintEdges & 0x1)
+    if ((paintEdges & 0x1))
```

Comment:
These extra parentheses aren't necessary, and please replace the hardcoded numbers with `EDGE_NE`, `EDGE_SE`, `EDGE_SW` and `EDGE_NW`.